app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteAddFragment.kt

```diff
@@ -59,10 +61,14 @@ class AutocompleteAddFragment : Fragment() {
                     .toLowerCase()
                     .removePrefixesIgnoreCase("http://", "https://", "www.")
 
-            if (domain.isEmpty()) {
-                domainView.error = getString(R.string.preference_autocomplete_add_error)
-            } else {
-                saveDomainAndClose(activity.applicationContext, domain)
+            launch(UI) {
+                if (domain.isEmpty()) {
```

Comment:
This if/else expression looks like it could be cleaned up with a [when](https://kotlinlang.org/docs/reference/control-flow.html#when-expression) :)